cmdstanpy/utils.py

```diff
@@ -1211,24 +1225,40 @@ def install_cmdstan(
 
 def wrap_url_progress_hook() -> Optional[Callable[[int, int, int], None]]:
     """Sets up tqdm callback for url downloads."""
-    pbar: Any = tqdm(
-        unit='B',
-        unit_scale=True,
-        unit_divisor=1024,
-        colour='blue',
-        leave=False,
-    )  # type: ignore
-
-    def download_progress_hook(
-        count: int, block_size: int, total_size: int
-    ) -> None:
-        if pbar.total is None:
-            pbar.total = total_size
-            pbar.reset()
-        downloaded_size = count * block_size
-        pbar.update(downloaded_size - pbar.n)
-        if pbar.n >= total_size:
-            pbar.close()
+    try:
+        pbar: Any = tqdm(
+            unit='B',
+            unit_scale=True,
+            unit_divisor=1024,
+            colour='blue',
+            leave=False,
+        )  # type: ignore
+    # pylint: disable=broad-except
+    except Exception as e:
+        disable_progress(e)
+
+        def download_progress_hook(
+            # pylint: disable=unused-argument
+            count: int, block_size: int, total_size: int
+        ) -> None:
+            return
+
+    else:
+
+        def download_progress_hook(
+            count: int, block_size: int, total_size: int
+        ) -> None:
+            try:
+                if pbar.total is None:
+                    pbar.total = total_size
+                    pbar.reset()
+                downloaded_size = count * block_size
+                pbar.update(downloaded_size - pbar.n)
+                if pbar.n >= total_size:
+                    pbar.close()
+            # pylint: disable=broad-except
+            except Exception as e:
```

Comment:
Once again, this try/except seems like a bad idea to me. Try/excepts aren’t free operations and this will only get triggered if there’s an issue in our code. TQDM import errors should all happen during the constructor call

Reply:
it's not an import error - it's that Jupyter notebook doesn't have ipywidgets installed and enabled.  so things blow up.  can this be checked for?

Reply:
For me, if I try to use TQDM without ipywidgets, the failure occurs exactly when I call `tqdm()` and raises a python `ImportError`. What error are you getting?